timApp/document/editing/routes.py

```diff
@@ -328,6 +328,37 @@ def modify_paragraph_common(doc_id: int, md: str, par_id: str, par_next_id: str
         elif tr_opt:
             if p.is_translation():
                 deref = mark_as_translated(p)
+
+                # Since empty area_end paragraphs are not visible, we need to set their rt-attribute
+                # at the same time as the area start paragraph. Otherwise, the 'Translation out of date' and
+                # 'Check translation' markings will remain on the area_end pars with no obvious way to remove them.
+                # For new area sections, the classes 'troutofdate' and 'checktr' will not be set at all for the area_end
+                # paragraphs if it is empty.
+                if p.is_area():
+                    area_name = (
+                        p.get_attr("area")
+                        if p.get_attr("area")
+                        else p.get_attr("area_end")
+                    )
+                    if not area_name:
+                        raise RouteException(
+                            "Could not access paragraph attributes ['area', 'area_end']."
+                        )
```

Comment:
Tätä ei taida ikinä tapahtua? `p.is_area()` taitaa kuitenkin tarkistaa, että joko `area` tai `area_end` on määritelty.
Jos ymmärsin oikein, tämä lisättiin mypyn takia? Jos joo, niin ehkä syntaktisesti vähän selkeämpi olisi minusta assertin käyttö:

```suggestion
                    assert area_name is not None
```